### PR TITLE
feat: bump up harbor to 2.3

### LINKED ISSUE #647: Support Harbor 2.3
Prometheus extended to jobserivce.
CRD changes are needed.

### PR DESCRIPTION
**Pull request by heww**
Closes #647

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: apis/goharbor.io/v1beta1/exporter_types.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ type ExporterSpec struct {
 	// +kubebuilder:validation:Required
 	Database ExporterDatabaseSpec `json:"database"`
 
+	// +kubebuilder:validation:Required
+	JobService ExporterJobServiceSpec `json:"jobservice"`
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/exporter_types.go:60]: If this is a required field, there may be a problem going from v1alpha3 to v1beta1 and the initial value needs to be added to the convert

File: controllers/goharbor/jobservice/deployments.go [hunk 2]
```diff
@@ -232,14 +241,31 @@ func (r *Reconciler) GetDeployment(ctx context.Context, jobservice *goharborv1.J
...
+	port := httpPort
+	portName := harbormetav1.JobServiceHTTPPortName
+
 	if jobservice.Spec.TLS.Enabled() {
-		port = harbormetav1.JobServiceHTTPSPortName
+		port = httpsPort
+		portName = harbormetav1.JobServiceHTTPSPortName
+	}
+
+	containerPorts := []corev1.ContainerPort{{
```
Comment (cndoit18) [inline controllers/goharbor/jobservice/deployments.go:252]: Is it no longer supported to open both http and https ports?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -66,19 +66,19 @@ jobs:
...
-      REGISTRYCTL_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/harbor-registryctl:v2.2.1
-      REGISTRY_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/registry-photon:v2.2.1
-      CHARTMUSEUM_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/chartmuseum-photon:v2.2.1
-      NOTARYSERVER_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/notary-server-photon:v2.2.1
-      NOTARYSIGNER_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/notary-signer-photon:v2.2.1
-      TRIVY_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/trivy-adapter-photon:v2.2.1
-      HARBORCLUSTER_MINIO_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/minio:2021-04-06T23-11-00Z
-      HARBORCLUSTER_REDIS_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/redis:5.0-alpine
-      HARBORCLUSTER_POSTGRESQL_DOCKER_IMAGE_2_2_1: ghcr.io/goharbor/spilo-12:1.6-p3
+      CORE_DOCKER_IMAGE_2_3_0: ghcr.io/goharbor/harbor-core:v2.3.0
```
Comment (steven-zou) [inline .github/workflows/tests.yml:69]: It seems we should provide repository and tag suffix ENVs to simplify the image path settings here.

**Review by steven-zou — COMMENTED** [review]
@heww see my comment?

File: .github/workflows/tests.yml (hunk 3 above)
Comment (heww) [reply .github/workflows/tests.yml:69]: Issue created https://github.com/goharbor/harbor-operator/issues/701 and will be fixed in another PR.

File: controllers/goharbor/jobservice/deployments.go (hunk 2 above)
Comment (heww) [reply controllers/goharbor/jobservice/deployments.go:252]: Yes, it will only open the port the jobservice listened on.

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by cndoit18 — APPROVED** [review]
LGTM

**Pull request merged**